build.gradle.kts

```diff
@@ -0,0 +1,372 @@
+        classpath = sourceSets["integrationTest"].runtimeClasspath
+
+        configure<JacocoTaskExtension> {
+            // don"t instrument sdk, icons, ktlint, etc.
+            includes = listOf("software.aws.toolkits.*")
+            excludes = listOf("software.aws.toolkits.ktlint.*")
+        }
+
+        project.plugins.withId("org.jetbrains.intellij") {
+            systemProperty("log.dir", "${(project.extensions["intellij"] as org.jetbrains.intellij.IntelliJPluginExtension).sandboxDirectory}-test/logs")
+        }
+
+        mustRunAfter(tasks.test)
+    }
+
+    project.plugins.withId("org.jetbrains.intellij") {
+        tasks.withType(org.jetbrains.intellij.tasks.DownloadRobotServerPluginTask::class.java) {
+            this.version = remoteRobotVersion
+        }
+
+        tasks.withType(org.jetbrains.intellij.tasks.RunIdeForUiTestTask::class.java).all {
+            systemProperty("robot-server.port", remoteRobotPort)
+            systemProperty("ide.mac.file.chooser.native", "false")
+            systemProperty("jb.consents.confirmation.enabled", "false")
+            // This does some magic in EndUserAgreement.java to make it not show the privacy policy
+            systemProperty("jb.privacy.policy.text", "<!--999.999-->")
+            if (System.getenv("CI") != null) {
+                systemProperty("aws.sharedCredentialsFile", "/tmp/.aws/credentials")
+            }
+        }
+
+        extensions.getByType<JacocoPluginExtension>().applyTo(tasks.getByName<RunIdeForUiTestTask>("runIdeForUiTests"))
+    }
+
+    tasks.withType(KotlinCompile::class.java).all {
+        kotlinOptions.jvmTarget = "1.8"
+    }
+
+    // Force us to compile the integration tests even during check even though we don't run them
+    tasks.named("check") {
+        dependsOn.add(sourceSets.getByName("integrationTest").compileJavaTaskName)
+    }
+
+    val testJar = tasks.register<Jar>("testJar") {
+        baseName = "${project.name}-test"
+        from(sourceSets.test.get().output)
+        from(sourceSets.getByName("integrationTest").output)
+    }
+
+    artifacts {
+        add("testArtifacts", testJar)
+    }
+
+    // Remove the tasks added in by gradle-intellij-plugin so that we don"t publish/verify multiple times
+    project.afterEvaluate {
+        removeTask(tasks, org.jetbrains.intellij.tasks.PublishTask::class.java)
+        removeTask(tasks, org.jetbrains.intellij.tasks.VerifyPluginTask::class.java)
+        removeTask(tasks, org.jetbrains.intellij.tasks.BuildSearchableOptionsTask::class.java)
+    }
+}
+
+val ktlint: Configuration by configurations.creating
+
+inline fun <reified T : Task> removeTask(tasks: TaskContainer, takeType: Class<T>) {
+    tasks.withType(takeType).configureEach {
+        enabled = false
+    }
+}
+
+apply(plugin = "org.jetbrains.intellij")
+apply(plugin = "toolkit-change-log")
+
+(project.extensions.getByName("intellij") as IntelliJPluginExtension).apply {
```

Comment:
again could replace this with `project.extensions.getByType`

Reply:
replacing with a buildSrc function